### PR TITLE
[GITHUB-17] Adds type to application logs

### PR DESCRIPTION
**Pull request by lobsterdore**
Adds the ability to specify a type for application logs, this
allows Logstash to perform grok and other transformations on
third party application logs.

Also added documentation detailing each builtin shipper.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,10 +11,11 @@ language: python
 python: "2.7"
 
 env:
-  - ANSIBLE_INSTALL_VERSION=1.9.4
   - ANSIBLE_INSTALL_VERSION=2.0.2.0
-  - ANSIBLE_INSTALL_VERSION=2.1.1.0
-  - ANSIBLE_INSTALL_VERSION=2.2.0.0
```
Comment (racer1680) [inline .travis.yml]: You assume no one uses this version...

File: meta/main.yml [hunk 2]
```diff
@@ -3,13 +3,12 @@
...
-  author: "David Reed"
   description: "Install Rsyslog."
   license: MIT
-  min_ansible_version: 1.9
+  min_ansible_version: 2.0
   platforms:
     - name: Ubuntu
       versions:
-        - all
+        - trusty
```
Comment (sreddel) [inline meta/main.yml]: will this cause an install on xenial to fail, or is it just advisory?
we are using the startup scripts included in the deb, so presumably it should work ok on xenial?

File: .travis.yml (hunk 1 above)
Comment (lobsterdore) [reply .travis.yml]: I think it's a good idea to use the latest patch version of each major version.

File: meta/main.yml (hunk 2 above)
Comment (lobsterdore) [reply meta/main.yml]: It's just advisory, if this works on Xenial then I can add it to the list.

Comment (lobsterdore) [reply meta/main.yml]: Added Xenial

File: .travis.yml (hunk 1 above)
Comment (lobsterdore) [reply .travis.yml]: I don't we should test every patch, we'll end up testing sheds loads of versions. If you look at the recent updates these are the versions that we are testing. I agree that long term we should fine a way to centralise these versions, I open to suggestions on that.

Comment (racer1680) [reply .travis.yml]: We don't run tests manually :)

**Pull request merged**